model/simulationtests/coil_cooling_dx.rb

```diff
@@ -0,0 +1,194 @@
+cool_cap_fff = curve_quadratic(m, 1, 0, 0, 0, 2, 0, 2)
+cool_eir_ft = curve_biquadratic(m, 0.70183064548, -0.049235917464, 0.00126482472, 0.031408426368, 0.0003622673484, -0.0011050729236, 13.88, 23.88, 18.33, 51.66)
+cool_eir_fff = curve_quadratic(m, 1, 0, 0, 0, 2, 0, 2)
+cool_plf_fplr = curve_quadratic(m, 0.75, 0.25, 0, 0, 1, 0.7, 1)
+
+speed_3.setTotalCoolingCapacityModifierFunctionofTemperatureCurve(cool_cap_ft)
+speed_3.setTotalCoolingCapacityModifierFunctionofAirFlowFractionCurve(cool_cap_fff)
+speed_3.setEnergyInputRatioModifierFunctionofTemperatureCurve(cool_eir_ft)
+speed_3.setEnergyInputRatioModifierFunctionofAirFlowFractionCurve(cool_eir_fff)
+speed_3.setPartLoadFractionCorrelationCurve(cool_plf_fplr)
+speed_3.setWasteHeatModifierFunctionofTemperatureCurve(constant_biquadratic)
+
+speed_4 = OpenStudio::Model::CoilCoolingDXCurveFitSpeed.new(m)
+speed_4.setGrossTotalCoolingCapacityFraction(12937.4031800778 / 4015.05615933448)
+speed_4.setGrossSensibleHeatRatio(0.712483430089655)
+speed_4.setGrossCoolingCOP(4.0695894950265)
+
+cool_cap_ft = curve_biquadratic(m, 0.962249880784, -0.0039087090378, 0.00132934689648, 0.0016072565382, -8.342352e-008, -0.00065486142576, 13.88, 23.88, 18.33, 51.66)
+cool_cap_fff = curve_quadratic(m, 1, 0, 0, 0, 2, 0, 2)
+cool_eir_ft = curve_biquadratic(m, -0.74798782608, 0.099914996016, -0.00272789532, 0.026966547, 0.0002513297808, -0.0006019728516, 13.88, 23.88, 18.33, 51.66)
+cool_eir_fff = curve_quadratic(m, 1, 0, 0, 0, 2, 0, 2)
+cool_plf_fplr = curve_quadratic(m, 0.75, 0.25, 0, 0, 1, 0.7, 1)
+
+speed_4.setTotalCoolingCapacityModifierFunctionofTemperatureCurve(cool_cap_ft)
+speed_4.setTotalCoolingCapacityModifierFunctionofAirFlowFractionCurve(cool_cap_fff)
+speed_4.setEnergyInputRatioModifierFunctionofTemperatureCurve(cool_eir_ft)
+speed_4.setEnergyInputRatioModifierFunctionofAirFlowFractionCurve(cool_eir_fff)
+speed_4.setPartLoadFractionCorrelationCurve(cool_plf_fplr)
+speed_4.setWasteHeatModifierFunctionofTemperatureCurve(constant_biquadratic)
+
+# CoilCoolingDXCurveFitOperatingMode
+operating_mode = OpenStudio::Model::CoilCoolingDXCurveFitOperatingMode.new(m)
+operating_mode.setRatedGrossTotalCoolingCapacity(4015.05615933448)
+operating_mode.setMaximumCyclingRate(3)
+operating_mode.setRatioofInitialMoistureEvaporationRateandSteadyStateLatentCapacity(1.5)
+operating_mode.setLatentCapacityTimeConstant(45)
+operating_mode.setNominalTimeforCondensateRemovaltoBegin(1000)
+
+operating_mode.addSpeed(speed_1)
+operating_mode.addSpeed(speed_2)
+operating_mode.addSpeed(speed_3)
+operating_mode.addSpeed(speed_4)
+operating_mode.setNominalSpeedNumber(1)
+
+# CoilCoolingDXCurveFitPerformance
+performance = OpenStudio::Model::CoilCoolingDXCurveFitPerformance.new(m, operating_mode)
+
+# CoilCoolingDX
+coil = OpenStudio::Model::CoilCoolingDX.new(m, performance)
```

Comment:
Correctly tests for all 4 classes concerned:

* 1 `CoilCoolingDX`
* 1 `CoilCoolingDXCurveFitPerformance`
* 1 `CoilCoolingDXCurveFitOperatingMode`
* 2 `CoilCoolingDXCurveFitSpeed`

Good example of how to use the API for all these objects too :+1: